src/general/IWorldEffect.cs

```diff
@@ -36,20 +38,63 @@ public void OnTimePassed(double elapsed, double totalTimePassed)
     {
         var glucose = SimulationParameters.Instance.GetCompound("glucose");
 
+        var totalAmount = 0.0f;
+        var totalChunkAmount = 0.0f;
+
+        var initialTotalDensity = 0.0f;
+        var finalTotalDensity = 0.0f;
+
         foreach (var key in targetWorld.Map.Patches.Keys)
         {
             var patch = targetWorld.Map.Patches[key];
 
+            if (!patch.Biome.Compounds.TryGetValue(glucose, out EnvironmentalCompoundProperties glucoseValue))
+                return;
+
+            totalAmount += glucoseValue.Amount;
+            initialTotalDensity += glucoseValue.Density;
+            totalChunkAmount += patch.GetTotalChunkCompoundAmount(glucose);
+
             // If there are microbes to be eating up the primordial soup, reduce the milk
             if (patch.SpeciesInPatch.Count > 0)
             {
-                if (patch.Biome.Compounds.TryGetValue(glucose, out EnvironmentalCompoundProperties glucoseValue))
-                {
-                    glucoseValue.Density = Math.Max(glucoseValue.Density * Constants.GLUCOSE_REDUCTION_RATE,
-                        Constants.GLUCOSE_MIN);
-                    patch.Biome.Compounds[glucose] = glucoseValue;
-                }
+                var initialDensity = glucoseValue.Density;
+
+                glucoseValue.Density = Math.Max(glucoseValue.Density * Constants.GLUCOSE_REDUCTION_RATE,
+                    Constants.GLUCOSE_MIN);
+                patch.Biome.Compounds[glucose] = glucoseValue;
+
+                var initialGlucose = Math.Round(initialDensity * glucoseValue.Amount +
+                    patch.GetTotalChunkCompoundAmount(glucose), 3);
```

Comment:
Couldn't this be calculated on line 61 so that the initialDensity doesn't need to be in a variable?